### DISCUSSION TITLE
Moon given intermittent behaviour when running tasks

### DISCUSSION THREAD

**Discussion by snspinn** [moonrepo/moon#1597] [General]
Hi, loving this project so far but I wanted to share some niggles and sharp corners that I've found so far. I don't have specific examples right now but I am finding moon is not behaving consistently, at all times.

Some things I've noticed:
-  When running tasks under tags (i.e. `moon run '#tag-name:start`) not all tasks run and I have to stop and re-run the process/tagged tasks.
- In same scenario as above:
    - If tasks-under-tag-name-count < 3, task name is not prepended to the logging output. 
    - If >= 3, task name (colourised) gets prepended.
- Moon doesn't always find node. This only will ever affect the first attempt and if I re-run the task immediately, it will find the node runtime perfectly. I have various versions of node installed under proto and at least two different versions, used in different tasks.

I am also running into all sort of npm/yarn caching/node_modules hellscapes, but I don't think that moon is contributing much friction there. Just wondering if anyone has seen these, has suggestions, or are these already known issues?

Will reiterate, really enjoying moon so far (tried Makefiles in the past :sob: :sob:) and this is a far more usable tool.

If code and config examples are needed I can work on getting something ready for sharing.

```
> moon --version
moon 1.25.6
>  proto --version
proto 0.36.2
```

**Comment by milesj** [comment]
@snspinn Hey, thanks for giving moon a try. Let's try and resolve your issues!

> When running tasks under tags (i.e. moon run '#tag-name:start) not all tasks run and I have to stop and re-run the process/tagged tasks.

How many projects have the tag you are trying to run, and how many of those projects have a start task? 

Can you also run it with `--log trace` and paste the output.

> If >= 3, task name (colourised) gets prepended.

The prefix logic for this quite simple. They only show if there's more than 1 primary target.

> Moon doesn't always find node.

Are you using the node toolchain in moon?

moon simply relies on `PATH` lookup for binaries to work. We don't do anything special. If proto isn't in your `PATH`, then there's not much we can do about it.

> I am also running into all sort of npm/yarn caching/node_modules hellscapes, but I don't think that moon is contributing much friction there.

Yeah moon just runs npm/yarn/etc commands for you. That's about it.

**Reply by snspinn** [reply]
Hi. Apologies for leaving this hanging. I'm using the proto toolchain.

I'll get logs here soon.
